indico/modules/events/templates/forms/event_creation_form.html

```diff
@@ -99,3 +99,215 @@
         $('#{{ form.title.id }}').focus();
     });
 </script>
+
+{% if check_room_availability %}
+    <script>
+        var rbExcludedCategories = {{ rb_excluded_categories | tojson }}
+        var serverDefaultTimezone = '{{ indico_config.DEFAULT_TIMEZONE }}';
+        var previousRoomId;
+        var $currentMessage;
+
+        var startDt;
+        var endDt;
+        var category;
+        var roomData;
+        var timezone;
+        var category;
+
+        var $createBooking = $('#event-creation-create_booking');
+        var $availableMessage = $('#room-available');
+        var $availablePrebookMessage = $('#room-available-prebook');
+        var $conflictBookingMessage = $('#room-conflict-booking');
+        var $conflictPrebookingMessage = $('#room-conflict-prebooking');
+        var $conflictPrebookingPrebookMessage = $('#room-conflict-prebooking-prebook');
+        var $userBookingMessage = $('#room-user-booking');
+        var $userPrebookingMessage = $('#room-user-prebooking');
+        var $unbookableMessage = $('#room-unbookable');
+        var $cantBookMessage = $('#room-cant-book');
+
+        function setLectureTimes(occurrence){
+            startDt = moment(occurrence['date'] + " " + occurrence['time'], 'DD-MM-YYYY HH:mm');
+            endDt = moment(startDt).add(occurrence['duration'], 'minutes');
+        }
+
+        function initValues() {
+            var startDate = $('#event-creation-start_dt-date').val();
+            var startTime = $('#event-creation-start_dt-time').val();
+            var endDate = $('#event-creation-end_dt-date').val();
+            var endTime = $('#event-creation-end_dt-time').val();
+            var occurrences = $('#event-creation-occurrences').val();
+
+            roomData = JSON.parse($("#event-creation-location_data").val());
+            timezone = $('#event-creation-timezone').val();
+            category = JSON.parse($("#event-creation-category").val());
+
+            if (occurrences && occurrences.length === 1) {
```

Comment:
Are you sure that is gonna work? `occurrences` here will contain a string not an array. So this basically will never match at this moment since you are checking the length of the stringified value of the field.